sdk/batch/batch/src/batchSharedKeyCredentials.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+import {
+  Constants,
+  WebResource,
+  ServiceClientCredentials,
+  HttpHeaders,
+  HttpMethods
+} from "@azure/ms-rest-js";
+import { HmacSha256Sign } from "./hmacSha256";
+const url = require("url-parse");
```

Comment:
@bterlson - I am not able to use 
- `import url from "url-parse"`. This causes errors with rollup. 
```bash
[!] Error: 'default' is not exported by node_modules/url-parse/index.js
https://rollupjs.org/guide/en#error-name-is-not-exported-by-module-
esm/batchSharedKeyCredentials.js (8:7)
 6: import { Constants } from "@azure/ms-rest-js";
 7: import { HmacSha256Sign } from "./hmacSha256";
 8: import url from "url-parse";
           ^
 9: /**
10:  * Creates a new BatchSharedKeyCredentials object.
Error: 'default' is not exported by node_modules/url-parse/index.js
    at error (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:3460:30)
    at Module.error (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:13358:9)
    at handleMissingExport (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:13048:21)
    at Module.traceVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:13466:17)
    at ModuleScope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:12713:29)
    at FunctionScope.Scope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:4022:68)
    at Scope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:4022:68)
    at FunctionScope.Scope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:4022:68)
    at Scope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:4022:68)
    at Identifier$$1.bind (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:10111:40)
```

- `import jssha from "jssha"`

  - warning
```bash
./esm/batchIndex.js → ./dist/batch.js...
(!) `this` has been rewritten to `undefined`
https://rollupjs.org/guide/en#error-this-is-undefined
node_modules/jssha/src/sha.js
43: 3733110249),new b(d[62],2999351573),new b(d[63],3815920427),new b(3391569614,3928383900),new b(3515267271,566280711),new b(3940187606,3454069534),new b(4118630271,4000239992),new b(116418474,1914138554),new b(174292421,2731055270),new b(289380356,3203993006),new b(460393269,320620315),new b(685471733,587496836),new b(852142971,1086792851),new b(1017036298,365543100),new b(1126000580,2618297676),new b(1288033470,3409855158),new b(1501505948,4234509866),new b(1607167915,987167468),new b(1816402316,
44: 1246189591)];X=[new b(0,1),new b(0,32898),new b(2147483648,32906),new b(2147483648,2147516416),new b(0,32907),new b(0,2147483649),new b(2147483648,2147516545),new b(2147483648,32777),new b(0,138),new b(0,136),new b(0,2147516425),new b(0,2147483658),new b(0,2147516555),new b(2147483648,139),new b(2147483648,32905),new b(2147483648,32771),new b(2147483648,32770),new b(2147483648,128),new b(0,32778),new b(2147483648,2147483658),new b(2147483648,2147516545),new b(2147483648,32896),new b(0,2147483649),
45: new b(2147483648,2147516424)];W=[[0,36,3,41,18],[1,44,10,45,2],[62,6,43,15,61],[28,55,25,21,56],[27,20,39,8,14]];"function"===typeof define&&define.amd?define(function(){return C}):"undefined"!==typeof exports?("undefined"!==typeof module&&module.exports&&(module.exports=C),exports=C):Y.jsSHA=C})(this);
```

   - error
```bash
[!] Error: 'default' is not exported by node_modules/jssha/src/sha.js
https://rollupjs.org/guide/en#error-name-is-not-exported-by-module-
esm/hmacSha256.js (6:7)
4:  * license information.
5:  */
6: import jssha from "jssha";
          ^
7: /**
8:  * Describes the HmacSHA256Sign object.
Error: 'default' is not exported by node_modules/jssha/src/sha.js
    at error (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:3460:30)
    at Module.error (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:13358:9)
    at handleMissingExport (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:13048:21)
    at Module.traceVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:13466:17)
    at ModuleScope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:12713:29)
    at FunctionScope.Scope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:4022:68)
    at Scope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:4022:68)
    at FunctionScope.Scope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:4022:68)
    at Scope.findVariable (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:4022:68)
    at Identifier$$1.bind (/Users/amarz/sdk/azure-sdk-for-js/sdk/batch/batch/node_modules/rollup/dist/rollup.js:10111:40)
```

I tried adding `module.export.default = Url` and `module.export.default = jssha` in respective packages in the node_modules folder. However, that did not help.

Reply:
As these are CJS dependencies, sometimes you have to manually declare their exports. See for [example here](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/service-bus/rollup.base.config.js#L132).